src/cpp/core/system/System.cpp

```diff
@@ -103,99 +106,133 @@ std::string generateShortenedUuid()
 std::string s_programIdentity;
 
 // logging options representing the latest state of the logging configuration file
-boost::shared_ptr<LogOptions> s_logOptions;
+boost::shared_ptr<log::LogOptions> s_logOptions;
 
 // mutex for logging synchronization
 boost::recursive_mutex s_loggingMutex;
 
 namespace {
 
-// main log writer for the binary
-LogWriter* s_pLogWriter = nullptr;
-
-// lazy initalized named loggers
-// these are secondary loggers to the main binary logger
-// useful for isolating specific sections of code within the binary
-typedef std::map<std::string, boost::shared_ptr<LogWriter> > SecondaryLoggerMap;
-SecondaryLoggerMap s_secondaryLogWriters;
-
-// additional log writers
-// these are created entirely in code and are not modifiable by the
-// logging configuration file
-std::vector<boost::shared_ptr<LogWriter> > s_additionalLogWriters;
+static std::atomic_uint s_fileLogId(3);
+static std::shared_ptr<log::ILogDestination> s_stdErrLogDest;
```

Comment:
Is it safe for these to be statics? (if they have non-trivial destructors, they could cause crashes when the process stops depending on order of destruction)

Reply:
Hm, I _think_ we should be safe for these, but let me refactor this to be sure.